### ISSUE TITLE
items is twice to much for images

### ISSUE THREAD

**Issue by clankill3r** [basiljs/basil.js#29]
If I place rectangles for example it's fine.
If I place images i always have twice as much.

````
	var items = b.items(b.layer("s_images"));
	b.println("items.length: "+items.length);
````

I think it both reports the content and the frame.

**Comment by fabiantheblind** [comment]
You are right. It is counting the image as well.

    var items = b.items(b.layer("img"));
    b.println("b.items: "+items.length);
    $.writeln("pageItems: " + (b.doc()).layers.item("img").pageItems.length);
    $.writeln("allPageItems: " + (b.doc()).layers.item("img").allPageItems.length);

In [includes/structure.js line 216](https://github.com/basiljs/basil.js/blob/master/includes/structure.js#L216) we are using. `allPageItems`. This should be `pageItems`. We will have to test if a change would affect other parts where `b.items` is used.  

Possible problmes could exist in:  

- [b.ungroup in includes/environment.js line 379](https://github.com/basiljs/basil.js/blob/master/includes/environment.js#L379)

**Comment by trych** [comment]
I don't think this should cause any problems to switch to `pageItems`. With `b.ungroup()` it is currently returning `allPageItems`, I am not sure if this is really the desired behavior (as only the top level page Items were really grouped). We just should update the docs to tell the user that with `b.items()` only the top level items are treated.
